File: HollowHalfInvertedPyramid.py
k=int(input("Enter the length : "))

for i in range(k):
    for j in range(k-i):
        if (i==0 or i==k-1 or j==0 or j==k-i-1):
            print("* ",end=" ")
        
        else:
            print("  ",end=" ")

    print(" ")

    

File: HollowHalfInvertedPyramid.cpp
#include <iostream>
using namespace std;
int main()
{
    int i, j, k;
    cout << "Enter the length : " << endl;
    cin >> k;
    cout<<"pattern :"<<endl;
    for (i = 0; i < k; i++)
    {
        for (j = 0; j <k-i; j++)
        {
            if (i == 0 || j == 0 || j == k-i-1 || i == k - 1)
            {
                cout << "* ";
            }
            else
            {
                cout << "  "; // ! doo space jaurui hai verna pattern galat banega
            }
        }
        cout << endl;
    }
    return 0;
}